lib/connectors/memory.js

```diff
@@ -457,15 +457,13 @@ function applyFilter(filter) {
   }
 
   function test(example, value) {
-    if (typeof value === 'string' && (example instanceof RegExp)) {
-      return value.match(example);
-    }
-
-    if (example === undefined) {
+    if (example === undefined)
       return undefined;
-    }
 
     if (typeof example === 'object' && example !== null) {
+      if (example instanceof RegExp)
```

Comment:
Please revert the connector change now that you normalize the shortcut to full regexp operator.